internal/driver/fetch.go

```diff
@@ -396,39 +396,54 @@ func locateBinaries(p *profile.Profile, s *source, obj plugin.ObjTool, ui plugin
 		searchPath = filepath.Join(os.Getenv(homeEnv()), "pprof", "binaries")
 	}
 mapping:
+
 	for _, m := range p.Mapping {
 		var baseName string
 		if m.File != "" {
 			baseName = filepath.Base(m.File)
 		}
-
-		for _, path := range filepath.SplitList(searchPath) {
-			var fileNames []string
-			if m.BuildID != "" {
-				fileNames = []string{filepath.Join(path, m.BuildID, baseName)}
+		var fileNames []string
+		if m.BuildID != "" {
+			var debugInfoName string
+			// gdb-style debuginfo expects a 160-bit sha-1 string as a build id.
+			// Look in paths like .build-id/ab/cdef0123456.debug
+			if len(m.BuildID) == 40 {
+				debugInfoName = filepath.Join(".build-id", m.BuildID[:2], m.BuildID[2:]+".debug")
+			}
+			if runtime.GOOS != "windows" && debugInfoName != "" {
+				// Standard global gdb search path.
+				fileNames = append(fileNames, filepath.Join("/usr/lib/debug", debugInfoName))
+			}
+			for _, path := range filepath.SplitList(searchPath) {
+				fileNames = append(fileNames, filepath.Join(path, m.BuildID, baseName))
 				if matches, err := filepath.Glob(filepath.Join(path, m.BuildID, "*")); err == nil {
 					fileNames = append(fileNames, matches...)
 				}
 				fileNames = append(fileNames, filepath.Join(path, m.File, m.BuildID)) // perf path format
+				if runtime.GOOS != "windows" && debugInfoName != "" {
```

Comment:
It might be easier to read the whole gdb search path support as a separate condition and loop:

```go
searchPaths := filepath.SplitList(searchPath)
...
if runtime.GOOS != "windows" && len(m.BuildID) == 40 {
  // gdb-style debuginfo expects a 160-bit sha-1 string as a build id.
  // Look in paths like .build-id/ab/cdef0123456.debug
  debugInfoName := filepath.Join(".build-id", m.BuildID[:2], m.BuildID[2:]+".debug")
  for _, path := range append(searchPaths, "/usr/lib/debug") {
    fileNames = append(fileNames, filepath.Join(path, debugInfoName)) // debuginfo format
  }
}
```